### ISSUE TITLE
http binding - route property

### ISSUE THREAD

**Issue by bartlomiejmont** [willmorgan/stub-azure-function-context#41]
https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook-trigger?tabs=python-v2%2Cin-process%2Cfunctionsv2&pivots=programming-language-javascript#customize-the-http-endpoint

```json
  {
      "bindings": [
      {
          "type": "httpTrigger",
          "name": "req",
          "direction": "in",
          "methods": [ "get" ],
          "route": "products/{category:alpha}/{id:int?}"
      },
      {
          "type": "http",
          "name": "res",
          "direction": "out"
      }
      ]
  }
```

```js
  module.exports = async function (context, req) {
  
      var category = context.bindingData.category;
      var id = context.bindingData.id;
      var message = `Category: ${category}, ID: ${id}`;
  
      context.res = {
          body: message;
      }
  }
```

I've got simular code as in the link above but using `functionRunner` `context.bindingData.id;` remains undefined. is this supported in the lib ? It was working in the prevous version  ?

**Comment by dhensby** [comment]
Where's the code that calls the functionrunner? That's the bit where the data is populated so I suspect you aren't populating the parameters

**Comment by bartlomiejmont** [comment]
I've tried to pass body everywhere 
``` js
req: new HttpBinding({
	method: HTTP_METHOD,
	url: INCOMING_URL,
	headers: HEADERS,
	query: body,
	params: body,
	user: null,
	body,
}),
```

**Comment by bartlomiejmont** [comment]
In the past Trigger was assigning body to the context.bindingData can't find something simular now 
``` js
function addTrigger(context, definition, trigger) {
    const { name } = definition;
    const request = {
        data: 'http',
        http: trigger,
    };
    Object.assign(context.bindings, {
        [name]: trigger,
    });
    Object.assign(context.bindingData, {
        $request: request,
        ...trigger.body,
        query: trigger.query,
        headers: trigger.headers,
        req: request,
    });
    Object.assign(context, {
        req: trigger,
    });
}
``` 
(last alfa version)

**Comment by bartlomiejmont** [comment]
It is working only if I spread body in HTTPBinding like this : 
``` js
		req: new HttpBinding({
			method: HTTP_METHOD,
			url: INCOMING_URL,
			headers: HEADERS,
			query: body,
			params: body,
			user: null,
			body,
			...body,
		}),
```

EDIT: This also does not work, it is overwriten in the `createHttpRequest` function

**Comment by dhensby** [comment]
Why would your GET request have a body? That's fairly unusual.

The category and id values are in the url params, so should be there:

```js
const req = new HttpBinding({
	method: HTTP_METHOD,
	url: INCOMING_URL,
	headers: HEADERS,
	params: {
            category: CATEGORY,
            id: ID,
        },
});
```

**Comment by bartlomiejmont** [comment]
Sorry that was an example from the azure docs. I have POST request. And it doesn't change the fact that parameters category and id are not saved in the `context.bindingData`. They are in `context.bindingData.req.params` and this is not true for azure function.

The problem is that it doesn't matter what i put in the HTTPBinding I can not get context.bindingData.id to have any value

**Comment by dhensby** [comment]
🤔  - you're saying the params are not available on the actual request object in genuine function contexts? That seems like there's a massive flaw in the assertions of the functions typings microsoft releases, not to say a horrendous flaw in this library.

Luckily, I use functions in production and know that this is not true, params are absolutely assigned to the request object in genuine function contexts.

What this library might not now be doing is assigning all these different vars to the `bindingData` object, but frankly, that is a horrible way of doing it, consumers of function apps should use the request objects and not this idiosyncratic spread of data across the `bindingData` object because it's so indeterminate.

This library isn't trying to run a 100% emulator of function apps (just run the function app if you want that) but to mock a context and execute a compatible function with the the (sensible) parameters that are provided to it.

However, I'm happy to spread the params over the `bindingData` object if that helps us stay compliant with the function apps, but you shouldn't rely on it (IMO). I'm happy to accept a PR on this.

**Comment by bartlomiejmont** [comment]
Yes params are available on the actual request object just not directly in the bindingData. I don't know why but this is how it is handled in azure docs so I think it will be good to be consistant with that. I will create a PR for that.

**Comment by dhensby** [comment]
fixed in #42
